### PR TITLE
Add warning when running start without build

### PR DESCRIPTION
**Pull request by ntwcklng**
We assume if no .next dir exists, the project has not yet been builded.

### CODE REVIEW

**Comment by ntwcklng** [comment]
Maybe even better: if no .next dir exists, we trigger the `next build` command first and then continue with `next start`?

File: bin/next-start [hunk 1]
```diff
@@ -38,7 +39,14 @@ if (argv.help) {
 const dir = resolve(argv._[0] || '.')
 
 const srv = new Server({ dir })
-srv.start(argv.port)
+
+exists(resolve(process.cwd(), '.next')).then(exists => {
```
Comment (arunoda) [inline bin/next-start]: It should be `dir` not `process.cwd`

Comment (arunoda) [inline bin/next-start]: It should be `dir` not `process.cwd()`

File: bin/next-start [hunk 2]
```diff
@@ -38,7 +39,14 @@ if (argv.help) {
...
+  if (!exists) {
+    console.error(`> Error! The application should be compiled with \`next build\` first.`)
+    process.exit(0)
```
Comment (arunoda) [inline bin/next-start]: We need exit with a non-zero code.

File: bin/next-start [hunk 3]
```diff
@@ -38,7 +39,14 @@ if (argv.help) {
 const dir = resolve(argv._[0] || '.')
 
 const srv = new Server({ dir })
-srv.start(argv.port)
+
+exists(resolve(process.cwd(), '.next')).then(exists => {
+  if (!exists) {
+    console.error(`> Error! The application should be compiled with \`next build\` first.`)
```
Comment (arunoda) [inline bin/next-start]: How about a message like this:

Could not found the '.next' directory! Try building your app with 'next build' before starting the server.

cc @rauchg

**Review by arunoda — CHANGES_REQUESTED** [review]

**Comment by arunoda** [comment]
@ntwcklng we should not build the app like this. Revert 171b544

**Comment by ntwcklng** [comment]
Okay sorry, i was a bit too hasty. Could we make it somehow possible to run the build command first?

File: bin/next-start [hunk 4]
```diff
@@ -38,7 +39,14 @@ if (argv.help) {
 const dir = resolve(argv._[0] || '.')
 
 const srv = new Server({ dir })
-srv.start(argv.port)
+
+exists(resolve(dir, '.next')).then(exists => {
```
Comment (arunoda) [inline bin/next-start]: fs.exists is a deprecated API. Use `fs.exitsSync`.
Import if from `fs` directly.

File: yarn.lock [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
 # yarn lockfile v1
+
+
```
Comment (nkzawa) [inline yarn.lock]: Please don't update `yarn.lock` which is not relevant to this PR.

File: bin/next-start (hunk 4 above)
Comment (ntwcklng) [inline bin/next-start]: Okay, should i still use mz/fs for that?

Comment (arunoda) [inline bin/next-start]: No. Directly import from `fs`.

**Review by arunoda — APPROVED** [review]

**Comment by nkzawa** [comment]
Thank you !

**Pull request merged**